web/ui/react-app/src/PanelList.tsx

```diff
@@ -6,19 +6,23 @@ import Panel, { PanelOptions, PanelDefaultOptions } from './Panel';
 import { decodePanelOptionsFromQueryString, encodePanelOptionsToQueryString } from './utils/urlParams';
 import Checkbox from './Checkbox';
 
+export type MetricGroup = { title: string; items: string[] };
```

Comment:
`MetricGroup` seems like a misnomer since queries are not (always) metrics / metric names. How about `CompletionGroup` (here and in comparable variable names)?

However, I wonder if we really have to abstract it this far, since I think there will only be two autocompletion groups for the foreseeable future (and I'm for adding abstraction only when it becomes necessary).

If we kept it simpler and just had separate top-level `pastQueries` and `metricNames` properties in the `PanelListState` (i.e. not grouped under `metricGroups`), then it would also simplify the code that reacts to new queries being saved in the history - it wouldn't have to ever have to refer to the `metricNames` at all, since those only get set once when the page loads. It would also have to reach fewer levels deep (`this.state.metricGroups.queries.items` -> `this.state.pastQueries`). And then maybe even leave the specific string title for each group up to the expression input.

What do you think?

Reply:
I think is more flexible now. If you decided to add another group someday(who knows) it's gonna be easier than if you have separate properties. Eventually only the PanelsList will be changed. The only downside I can see of this approach is the long path to the items.
```this.state.metricGroups.queries.items VS this.state.pastQueries```

Reply:
I don't feel too strongly about it in this case, just generally over the years of maintaining code I've become more and more of a fan of YAGNI (https://www.martinfowler.com/bliki/Yagni.html), KISS, and avoiding premature abstraction. IMO it makes code easier to read/understand/keep correct before an abstraction is needed.

My guess is that we will likely not add a third category before we change to completely a different solution that can do context-sensitive autocompletion (like auto-completing label names in `by` clauses, auto-completing function names, etc.). And if we do, we can still change things then.

That's just to outline my general thinking about "this will be more flexible in the future" questions, though again I don't feel too strongly about this particular case. If you'd rather like to keep it like it is, it's ok too.

Reply:
Assuming you are keeping the structure as it is for now, could you still rename it to `CompletionGroup`, `AutocompleteGroup`, `SearchGroup`, or something like that, since queries are not metrics?